pkg/signature/signerverifier.go

```diff
@@ -50,6 +50,28 @@ func LoadSignerVerifier(privateKey crypto.PrivateKey, hashFunc crypto.Hash) (Sig
 	return nil, errors.New("unsupported public key type")
 }
 
+// LoadSignerVerifierWithOpts returns a signature.SignerVerifier based on the
+// algorithm of the private key provided and the user's choice.
+func LoadSignerVerifierWithOpts(privateKey crypto.PrivateKey, hashFunc crypto.Hash, opts ...SignerVerifierOption) (SignerVerifier, error) {
```

Comment:
```suggestion
func LoadSignerVerifierWithOpts(privateKey crypto.PrivateKey, opts ...SignerVerifierOption) (SignerVerifier, error) {
```

As I had to add `opts ...SignerVerifierOption` to a bunch of methods in cosign, I wonder whether it makes sense to consider the hashFunc just part of the options, so that it can be passed more easily around.

Right now, crypto.SHA256 is hard-coded in many places and making that configurable would require adding a `hashFunc` argument to other functions (plus the SignerVerifierOption), so maybe we could just include the hash there.

What do you think?

Reply:
I added a commit with this change. You can also see in https://github.com/trail-of-forks/cosign/compare/support-ed25519ph...trail-of-forks:cosign:signing-algorithm-flag?expand=1 how it might be used in cosign.

Reply:
The downside of that would be having to handle when a hash isn't provided. `isSupportedAlg` will catch if no hash is passed, but it's an incorrect error message ("incorrect hash passed" rather than "no hash passed").

Reply:
Ah I missed https://github.com/sigstore/sigstore/pull/1595/files#diff-caa770c16c3f35a06be96442a7e99690a11c9ff12efbffa523e054dbbed29a10R98 - This covers my concern. I thought we didn't have a fallback.